### PR TITLE
Added case for VRF terminal unit in SystemReports

### PR DESCRIPTION
**Pull request by rraustad**
Copied B Griffith code for zone forced air unit mechanical ventilation reporting.
https://github.com/NREL/EnergyPlus/issues/4706

### CODE REVIEW

File: src/EnergyPlus/SystemReports.cc [hunk 1]
```diff
@@ -3994,108 +3999,145 @@ namespace SystemReports {
 
 			// first deal with any (and all) Zone Forced Air Units that might have outside air.
 			for ( thisZoneEquipNum = 1; thisZoneEquipNum <= ZoneEquipList( ZoneEquipConfig( CtrlZoneNum ).EquipListIndex ).NumOfEquipTypes; ++thisZoneEquipNum ) {
-				{ auto const SELECT_CASE_var( ZoneEquipList( ZoneEquipConfig( CtrlZoneNum ).EquipListIndex ).EquipType_Num( thisZoneEquipNum ) );
```
Comment (rraustad) [inline src/EnergyPlus/SystemReports.cc]: I don't understand this shift to the right.

**Comment by Myoldmopar** [comment]
@rraustad I did some minor cleanup on this branch.  Didn't touch any functionality.  I'll do some more technical review on it soon, probably tomorrow.

**Comment by rraustad** [comment]
Thanks. The biggest question is what to do at the end of the if block. @mjwitte had suggested some form of assert instead of testing all remaining equipment types. The primary reason for the last else if is for developers to notice that new models need to be accounted for here. Is there is some way to do this more efficiently?

**Comment by Myoldmopar** [comment]
That last merge from develop (beef588) should clean up the regression warnings.  I'll take a look at the if-block / assertion you mentioned in the comment and see if I can come up with something.

**Comment by Myoldmopar** [comment]
So an assertion would only be triggered on a debug build.  I think having the diagnostic fatal error in there will ensure that nothing ever sneaks past, even if the dev accidentally builds release.  We already have precedent for diagnostic fatals, so I'd say leave it and let's merge this.  Is there anything else looming before it is merge-able?

**Comment by rraustad** [comment]
This is good to go. I still have to add in a UnitarySystem (the blank if 
block just above the end) call when used as zone equipment but I'll get 
to that soon. And I wasn't sure if 3? of the other zone equipment types 
needed to be addressed (I put comments in the last else if block to 
highlight).

On 2/12/2015 4:55 PM, Edwin Lee wrote:
>
> So an assertion would only be triggered on a debug build. I think 
> having the diagnostic fatal error in there will ensure that nothing 
> ever sneaks past, even if the dev accidentally builds release. We 
> already have precedent for diagnostic fatals, so I'd say leave it and 
> let's merge this. Is there anything else looming before it is merge-able?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/NREL/EnergyPlus/pull/4707#issuecomment-74161469>.
>

-- 
Richard Raustad
Florida Solar Energy Center
1679 Clearlake Road
Cocoa, FL 32922
Ph: (321)638-1454
http://www.fsec.ucf.edu/en/

Program Director
Electric Vehicle Transportation Center
http://evtc.fsec.ucf.edu/

**Comment by Myoldmopar** [comment]
So for right now, if you put in one of those 3 (ZoneHVAC:EvaporativeCoolerUnit, eg.) as zone equipment and get that report, it will hit that fatal line?  

 - If so:
   - then side question: should the example files cover that configuration?  
 - If not:
   - ...I'm not sure.

**Comment by rraustad** [comment]
No, any of those 3 will fall through the last else if and do nothing. The question is, for any of those 3 zone equipment types (which look like they could have OA), should a calculation be made for OA (i.e. one or more else if's above to do something)? For example, when ZoneHVAC:OutdoorAirUnit was originally added, shouldn't this code have been modified to report the OA amount? Is that handled in another area of the code? I didn't run multiple test cases to test each of these, I only fixed what was wrong with the defect file.

**Comment by Myoldmopar** [comment]
What would be awesome is if there was a base class for zone equipment that required each sub class to implement a getOA function. Then equipment that didn't have OA would just return zero. Developers would have to be explicit about defining either a real function or returning zero. Future...

**Comment by mjwitte** [comment]
@rraustad Certainly seems that the zonehvac:outdoorairunit should be part of this.  There are two parts to this, there's the OA reporting (which is critical) and there's the OA energy use reports (which were always problematic).  Does this code feed into both of those?

**Comment by rraustad** [comment]
Used in these report variables:

Zone Mechanical Ventilation No Load Heat Removal Energy [J]
Zone Mechanical Ventilation Cooling Load Increase Energy [J]
Zone Mechanical Ventilation Cooling Load Increase Due to Overheating Energy [J
Zone Mechanical Ventilation Cooling Load Decrease Energy [J
Zone Mechanical Ventilation No Load Heat Addition Energy [J]
Zone Mechanical Ventilation Heating Load Increase Energy [J]
Zone Mechanical Ventilation Heating Load Increase Due to Overcooling Energy [J]
Zone Mechanical Ventilation Heating Load Decrease ]Energy [J]
Zone Mechanical Ventilation Mass Flow Rate [kg/s]
Zone Mechanical Ventilation Mass [kg]
Zone Mechanical Ventilation Standard Density Volume Flow Rate [m3/s]
Zone Mechanical Ventilation Standard Density Volume [m3]
Zone Mechanical Ventilation Current Density Volume Flow Rate [m3/s]
Zone Mechanical Ventilation Current Density Volume [m3]
Zone Mechanical Ventilation Air Changes per Hour [ach]

Where is the OA energy use report?

**Comment by rraustad** [comment]
I guess these report variables are both [J] = OA energy, others = OA reporting

**Comment by rraustad** [comment]
Mechanical ventilation = 0 in Outdoor Air Summary report for OutdoorAirUnit.idf.

**Comment by mjwitte** [comment]
@rraustad If mechanical ventilation = 0, then that needs to be fixed.  Regarding your question "where is OA energy use report", I'm fairly certain that code was pulled a while back, but we kept the mechanical ventilation load outputs that you list above.

**Comment by rraustad** [comment]
OutdoorAirUnit.idf now shows Mechanical Ventilation in Outdoor Air Summary Reports. It's odd that 0.42 m3/s (or 0.84 m3/s for Space5-1 that has 2 ZoneHVAC:OutdoorAir units) * 3600 divided by zone area does not quite equal ACH in this report. Is ACH based on mass or volume flow rate?

**Comment by rraustad** [comment]
Re: ACH not matching, the reports are based on current zone air density.
ZonePreDefRep( ActualZoneNum ).MechVentVolTotal += ZoneOAVolCrntRho( CtrlZoneNum );
RhoAir ~ 1.17 in my run.

The OA unit is initialized to standard density.
RhoAir = StdRhoAir; = ~ 1.2
OutAirUnit( OAUnitNum ).OutAirMassFlow = RhoAir * OAFrac * OutAirUnit( OAUnitNum ).OutAirVolFlow;

That's only about 3% of a difference, which from memory doesn't account for the difference I saw. The reports purportedly show the average for occupied hours. The occupancy schedule shows what would be an average of about 72% of the design input. I still haven't identified the difference between user input and reported value.
  Schedule:Compact,
    BLDG Sch   1,            !- Name
    Any Number,              !- Schedule Type Limits Name
    Through: 12/31,          !- Field 1
    For: Weekdays SummerDesignDay WinterDesignDay, !- Field 2
    Until: 6:00,0.0,         !- Field 3
    Until: 7:00,0.1,         !- Field 5
    Until: 8:00,0.5,         !- Field 7
    Until: 12:00,1.0,        !- Field 9
    Until: 13:00,0.5,        !- Field 11
    Until: 16:00,1.0,        !- Field 13
    Until: 17:00,0.5,        !- Field 15
    Until: 18:00,0.1,        !- Field 17
    Until: 24:00,0.0,        !- Field 19
    For: AllOtherDays,       !- Field 21
    Until: 24:00,0.0;        !- Field 22

**Comment by Myoldmopar** [comment]
@rraustad Based on your last comment, I'm guessing this still needs a little more investigation before proceeding toward a merge?  Or is that issue unrelated and a new issue should be opened to remedy that?

**Comment by rraustad** [comment]
I'd say the underlying issue of missing data in the Outdoor Air Summary report has been fixed. If there is an issue with the value reported, that is a separate issue. I vote merge.

**Comment by Myoldmopar** [comment]
OK, looking at the results, everything seems OK with this branch.  I'm going to re-merge develop and see if it goes more green.  This should merge after that. 

@rraustad If there is a value reporting issue, could you open a new GitHub issue for that?

**Comment by Myoldmopar** [comment]
Alright, there's the merge.  I think CI should be cleaner now, and this should be merge-able later today.

**Comment by rraustad** [comment]
I will open reporting issues as I run across them.

**Comment by Myoldmopar** [comment]
@rraustad CI is great now.  I don't want to be responsible for turning the develop branch non-green, so I'll let CI finish up a couple more tests and merge this.  Thanks.

**Pull request merged**